Sources/FoundationEssentials/FileManager/FileOperations.swift

```diff
@@ -788,7 +788,7 @@ enum _FileOperations {
         try src.withNTPathRepresentation { pwszSource in
             var faAttributes: WIN32_FILE_ATTRIBUTE_DATA = .init()
             guard GetFileAttributesExW(pwszSource, GetFileExInfoStandard, &faAttributes) else {
-                throw CocoaError.copyFileError(GetLastError(), src, dst)
+                throw CocoaError.errorWithFilePath(.fileReadNoSuchFile, src)
```

Comment:
Thanks for updating, this looks like a better location to address the issue. As for which error to throw, `copyFileError` attaches a handful of extra keys to the user info. The unit tests might not test for them specifically but for consistency, should we make a variant of `copyFileError` that takes a `CocoaError.Code` instead of an errno which adds the extra keys like the variant / destination?

Reply:
I guess I'm missing what the consistency would be with. I don't really understand what the expectation is here, so I'm just relying on the tests to tell me if the error object is properly constructed.

Reply:
It'd be consistency with the behavior across other platforms. You're right today we don't have a unit test for that. We have

```
XCTAssertThrowsError(try $0.copyItem(atPath: "does_not_exist", toPath: "dir3")) {
    XCTAssertEqual(($0 as? CocoaError)?.code, .fileReadNoSuchFile)
}
```

But that only tests the code and not the other keys like the source/destination file and the variant. We could probably update the unit test to account for those but I think the expectation here in my mind is that it throws the same error as Darwin/Linux which both currently set the variant to "Copy" and provide source/destination paths in the user info. It might not be critical or load bearing, but since we're already working here to try to maintain consistent behavior do you think we should update this to throw the same error contents in entirety rather than losing the other contents while getting the code right?

Reply:
Hmm, that type of uniformity is definitely preferable. I can look into adding the extra initialiser for that.